vnext/Microsoft.ReactNative.Cxx.UnitTests/NativeModuleTest.cpp

```diff
@@ -772,4 +775,24 @@ TEST_CASE_METHOD(NativeModuleTestFixture, "TestEvent_EventField2", "NativeModule
   REQUIRE(eventRaised == true);
 }
 
+TEST_CASE_METHOD(NativeModuleTestFixture, "TestEvent_EventField3", "NativeModuleTest") {
+  bool eventRaised = false;
+  m_builderMock.SetEventHandler(
+      L"OnObjectResult3", std::function<void(const JSValue &)>([&eventRaised](const JSValue &eventArg) noexcept {
+        REQUIRE(eventArg.Object().at("X").Int64() == 4);
+        REQUIRE(eventArg.Object().at("Y").Int64() == 2);
+        eventRaised = true;
+      }));
+
+  JSValue data = JSValue();
+  auto writer = MakeJSValueTreeWriter(data);
+  writer.WriteObjectBegin();
+  WriteProperty(writer, "X", 4);
+  WriteProperty(writer, "Y", 2);
+  writer.WriteObjectEnd();
```

Comment:
Consider using more compact syntax which should be also more efficient:

JSValue data{ JSValueObject {
  {"X", 4},
  {"Y", 2}
}};